client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js

```diff
@@ -21,153 +21,297 @@ import SetupHeader from './setup-header';
+			'wc-api-dev': translate( 'WooCommerce API Dev' ),
+			'woocommerce-gateway-stripe': translate( 'WooCommerce Stripe Gateway' ),
+			'woocommerce-services': translate( 'WooCommerce Services' ),
+			'taxjar-simplified-taxes-for-woocommerce': translate( 'TaxJar - Sales Tax Automation for WooCommerce' ),
+		};
+	}
+
+	doInitialization = () => {
+		const { site, sitePlugins, translate, wporg } = this.props;
+
+		if ( ! site ) {
 			return;
 		}
-		const isReady = plugins.length && ! this.state.installingPlugin && ! this.state.activatingPlugin;
-		const isDoneInstalling = prevProps.plugins && plugins.length > prevProps.plugins.length;
-		const activatingPlugin = find( plugins, { slug: this.state.activatingPlugin } );
-		const isDoneActivating = activatingPlugin && activatingPlugin.active;
-		if (
-			isReady ||
-			( isDoneInstalling || isDoneActivating )
-		) {
-			this.installPlugins( this.props.plugins );
+
+		let waitingForPluginListFromSite = false;
+		if ( ! sitePlugins ) {
+			waitingForPluginListFromSite = true;
+		} else if ( ! Array.isArray( sitePlugins ) ) {
+			waitingForPluginListFromSite = true;
+		} else if ( 0 === sitePlugins.length ) {
+			waitingForPluginListFromSite = true;
 		}
-	}
 
-	getWporgPluginData() {
-		requiredPlugins.map( plugin => {
-			if ( 'wc-api-dev' === plugin ) {
-				return;
-			}
-			const pluginData = getPlugin( this.props.wporg, plugin );
+		if ( waitingForPluginListFromSite ) {
+			this.setState( {
+				message: translate( 'Waiting for plugin list from site' ),
+				progress: 0,
+			} );
+			return;
+		}
+
+		// Iterate over the required plugins, fetching plugin
+		// data from wordpress.org for each into state
+		const requiredPlugins = this.getRequiredPluginsList();
+		let pluginDataLoaded = true;
+		for ( const requiredPluginSlug in requiredPlugins ) {
+			const pluginData = getPlugin( wporg, requiredPluginSlug );
+			// pluginData will be null until the action has had
+			// a chance to try and fetch data for the plugin slug
+			// given. Note that non-wp-org plugins (like wc-api-dev)
+			// will be accepted too, but with
+			// { fetched: false, wporg: false }
+			// as their response
 			if ( ! pluginData ) {
-				this.props.fetchPluginData( plugin );
+				this.props.fetchPluginData( requiredPluginSlug );
+				pluginDataLoaded = false;
 			}
-		} );
-	}
+		}
+		if ( ! pluginDataLoaded ) {
+			this.setState( {
+				message: translate( 'Loading plugin data' ),
+				progress: 0,
+			} );
+			return;
+		}
 
-	installApiDevPlugin = ( siteId ) => {
-		const progress = this.state.progress + ( 100 / requiredPlugins.length );
-		this.setState( () => ( {
-			installingPlugin: 'wc-api-dev',
-			progress,
-		} ) );
+		const toInstall = [];
+		const toActivate = [];
+		for ( const requiredPluginSlug in requiredPlugins ) {
+			const pluginFound = find( sitePlugins, { slug: requiredPluginSlug } );
+			if ( ! pluginFound ) {
+				toInstall.push( requiredPluginSlug );
+				toActivate.push( requiredPluginSlug );
```

Comment:
> Unless the api-dev plugin endpoint doesn't also activate it

The API plugin enables auto-update on the .org side https://github.com/woocommerce/wc-api-dev/blob/master/auto-update.php#L24, since we have to hook into the update system to do things correctly from GitHub.

I think this probably does nothing in that case, so it seems like it could be removed. The Calypso UI currently says disabled with a message "This plugin is not in the WordPress.org plugin repository, so we can't autoupdate it." no matter what -- which is something we should look into fixing independent of this PR (not a blocker).